convert.c

```diff
@@ -969,6 +1099,32 @@ static int ident_to_worktree(const char *path, const char *src, size_t len,
 	return 1;
 }
 
+static struct encoding *git_path_check_encoding(struct attr_check_item *check)
```

Comment:
I wouldn’t necessarily call this a `check` function, because it does way more than that. First, the output of the function is the encoding, so it’s more or less a getter function. Second, it’s lazy initialization in that it creates missing encodings on the fly.

An idiom for naming such functions is `get_or_create`, so maybe, this could be renamed to `git_path_get_or_create_encoding` or so.

Reply:
I guess this follows the other function names in the file ;-)